Kernel/Net/NetworkTask.cpp

```diff
@@ -276,7 +280,25 @@ void handle_udp(const IPv4Packet& ipv4_packet, const Time& packet_timestamp)
     auto& destination = ipv4_packet.destination();
 
     if (destination == IPv4Address(255, 255, 255, 255) || NetworkAdapter::from_ipv4_address(destination) || socket->multicast_memberships().contains_slow(destination))
-        socket->did_receive(ipv4_packet.source(), udp_packet.source_port(), KBuffer::copy(&ipv4_packet, sizeof(IPv4Packet) + ipv4_packet.payload_size()), packet_timestamp);
+        socket->did_receive(ipv4_packet.source(), udp_packet.source_port(), { &ipv4_packet, sizeof(IPv4Packet) + ipv4_packet.payload_size() }, packet_timestamp);
+}
+
+void queue_tcp_ack(RefPtr<TCPSocket> socket)
+{
+    if (!delayed_ack_sockets)
+        delayed_ack_sockets = new HashTable<RefPtr<TCPSocket>>;
```

Comment:
We should avoid attempting to lazily allocate stuff like this.. it's racy and there is no way to propagate potential OOM with this API.

Reply:
This only happens in `NetworkTask` though, so there's no race. Do you have a better idea? :)

Reply:
> This only happens in `NetworkTask` though, so there's no race. 

Sorry, I thought I saw this being called from TcpSocket.

> Do you have a better idea? :)

This should get pre-allocated in the network task factory, which should either assert success / return failure on boot. 
There is a ton of other stuff to fix in NetworkTask though, so I'm not too worried about it... just trying to shift peoples
mental model to to not assume allocation success.

Reply:
Cool, I've moved the initialization to the start of the `NetworkTask_main` function. Once `std::nothrow` hits the master branch we can update that. :+1: